sdk/keyvault/azure-security-keyvault-perf/src/main/java/com/azure/security/keyvault/secrets/perf/ListSecretsTest.java

```diff
@@ -10,16 +10,43 @@
 import java.util.UUID;
 
 public class ListSecretsTest extends SecretsTest<PerfStressOptions> {
+    private String[] _secretNames;
+
     public ListSecretsTest(PerfStressOptions options) {
         super(options);
     }
 
+    @Override
     public Mono<Void> globalSetupAsync() {
-        return super.globalSetupAsync()
-            .then(Flux.range(0, options.getCount())
-                .map(i -> "listSecretsPerfTest-" + UUID.randomUUID())
-                .flatMap(b -> secretAsyncClient.setSecret(b, b))
-                .then());
+        super.globalSetupAsync().block();
```

Comment:
We should avoid calling `block()` in an async method.

Reply:
Do you know how to convert the code in this method to reactor?  I used `block()` here because I wasn't sure how to chain all the other code after `globalSetupAsync()`.  One way I would know how to do it, is to move all the code after `globalSetupAsync()` into a separate async method, then chain this after `globalSetupAsync()`.

Reply:
I attempted to convert it using `then(Mono.defer())`.  Is this the best way?